### PR TITLE
[doc] Move ICU documentation to asciidoc

### PR DESCRIPTION
**Pull request by dadoonet**
This commit moves `plugins/analysis-icu/README.md` file to `plugins/analysis-icu/src/main/asciidoc/index.asciidoc`.

It will be easier to integrate plugin documentation in our reference guide.

@clintongormley follow up #11630

### CODE REVIEW

File: plugins/analysis-icu/src/main/asciidoc/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+`你好` to `ni hao`
+
+`здравствуйте` to `zdravstvujte`
+
+`こんにちは` to `kon'nichiha`
+
+Currently the filter only supports identifier and direction, custom rulesets are not yet supported.
+
+For more documentation, Please see the [user guide of ICU Transform](http://userguide.icu-project.org/transforms/general).
```
Comment (clintongormley) [inline plugins/analysis-icu/src/main/asciidoc/index.asciidoc]: bad link formatting here

File: plugins/analysis-icu/src/main/asciidoc/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,307 @@
+[[analysis-icu]]
+= ICU Analysis for Elasticsearch
```
Comment (clintongormley) [inline plugins/analysis-icu/src/main/asciidoc/index.asciidoc]: ==

File: plugins/analysis-icu/src/main/asciidoc/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+[source,shell]
+--------------------------------------------------
+mvn clean install
+plugin --install analysis-icu \
+       --url file:target/releases/elasticsearch-analysis-icu-X.X.X-SNAPSHOT.zip
+--------------------------------------------------
+
+
+[analysis-icu-normalization]
+== ICU Normalization
```
Comment (clintongormley) [inline plugins/analysis-icu/src/main/asciidoc/index.asciidoc]: ===

File: plugins/analysis-icu/src/main/asciidoc/index.asciidoc [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                    "filter" : ["icu_normalizer"]
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+
+[analysis-icu-folding]
+== ICU Folding
```
Comment (clintongormley) [inline plugins/analysis-icu/src/main/asciidoc/index.asciidoc]: ===

File: plugins/analysis-icu/src/main/asciidoc/index.asciidoc [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+                    "filter" : ["icu_folding"]
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+
+[analysis-icu-filtering]
+== ICU Filtering
```
Comment (clintongormley) [inline plugins/analysis-icu/src/main/asciidoc/index.asciidoc]: ===

etc for all the other headers

**Comment by clintongormley** [comment]
Also, please add the installation/removal/usage sections

**Comment by dadoonet** [comment]
PR updated @clintongormley

**Comment by dadoonet** [comment]
Closing in favor of #12040

**Pull request closed without merging**